auth/credentials/transport.go

```diff
@@ -0,0 +1,27 @@
+package credentials
+
+import (
+	"fmt"
+	"net/http"
+)
+
+// OAuthCustomHTTPTransport is a custom HTTP transport that injects the OAuth2 token into requests.
+type OAuthCustomHTTPTransport struct {
+	underlyingTransport http.RoundTripper
+	client              *OAuthClient
+}
+
+// RoundTrip implements the RoundTripper interface.
+func (t *OAuthCustomHTTPTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+    // Get a valid token (refreshing it if necessary)
+    token, err := t.client.getValidToken() // Get or refresh the token
```

Comment:
nit: indentation seems off

Reply:
Awesome find! Checking with golang ci lint is not failing for missing gofmt

Reply:
It looks like gofmt does not support wildcards in the format as other tools:

gofmt: The Go formatter tool.
-s: Simplifies the code by removing unnecessary whitespace and comments.
-w: Writes the changes back to the original files.
./**/**/*.go: A glob pattern that matches all .go files in the current directory and its subdirectories, recursively.


```
Projects/cloud/atlas-sdk-go  sa-poc ✔                                                                                                                                        5h5m  
▶ make fmt
==> Fixing source code with gofmt...
gofmt -s -w  ./**/*.go
// zero changes

▶ gofmt -s -w  ./**/**/*.go 
// All code formatted
``` 
🙈

Reply:
We also have disabled formatter on golang-ci-lint for formatter. Will get that enabled in separate PR.
@fmenezes thank you for spotting it. Reformatted source code and fixed make target.